core/src/Threads/Kokkos_Threads_Team.hpp

```diff
@@ -716,6 +715,10 @@ class TeamPolicyInternal<Kokkos::Threads, Properties...>
       : TeamPolicyInternal(typename traits::execution_space(),
                            league_size_request, team_size_request, -1) {}
 
+  TeamPolicyInternal(const PolicyUpdate, const TeamPolicyInternal& other,
+                     const typename traits::execution_space&)
+      : TeamPolicyInternal(other) {}
```

Comment:
(No action required)
Noting here as well that the team policy does not store its execution space.

Yuk
https://github.com/kokkos/kokkos/blob/b3ec45c33bc8151b92d63a764f01281911c83cf9/core/src/Threads/Kokkos_Threads_Team.hpp#L567-L570

Reply:
I noticed, and did not wanted to slow down this PR with such weird things :jack_o_lantern: 

But it clearly needs to be addressed.

Reply:
#8510